src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -395,26 +395,39 @@ private ISet<DocumentId> GetProjectOpenDocuments_NoLock(ProjectId project)
             return openDocs;
         }
 
-        internal Task OnDocumentOpenedAsync(
+        protected internal void OnDocumentOpened(
             DocumentId documentId, SourceTextContainer textContainer,
             bool isCurrentContext = true)
         {
-            OnDocumentOpened(documentId, textContainer, isCurrentContext);
-            return SpecializedTasks.EmptyTask;
+            OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext, wait: true).Wait(CancellationToken.None);
         }
 
-        protected internal void OnDocumentOpened(
+        internal Task OnDocumentOpenedAsync(
```

Comment:
It doesn't have to be part of this PR, I just don't want us to get complacent with unit tests testing a completely different path through here than product code.